### PR TITLE
Sync: Subscriptions do not work as expected.

### PR DESCRIPTION
**Pull request by enejb**
This test should pass since all the items in the queue should be independent independent of each other. This is currently not the case as this test demonstrates.

### CODE REVIEW

File: tests/php/sync/test_class.jetpack-sync-posts.php [hunk 1]
```diff
@@ -364,6 +364,40 @@ function test_sync_post_includes_dont_email_post_to_subs() {
 		$this->assertEquals( true, $post_on_server->dont_email_post_to_subs );
 	}
 
+
+	function test_sync_publicize_works_as_expected() {
```
Comment (enejb) [inline tests/php/sync/test_class.jetpack-sync-posts.php]: I do not. They both should have `$post->dont_email_post_to_subs` set to false since I would want the subscriptions to be sent out. In the case of the factory->post->create that post has a status of draft and so we would not be sending out the subscription for that anyways.

File: modules/subscriptions.php [hunk 2]
```diff
@@ -171,7 +171,7 @@ function maybe_send_subscription_email( $new_status, $old_status, $post ) {
 		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
 			return;
 		}
-
+
```
Comment (lezama) [inline modules/subscriptions.php]: we are introducing some whitespaces here

File: tests/php/sync/test_class.jetpack-sync-posts.php [hunk 3]
```diff
@@ -364,6 +364,48 @@ function test_sync_post_includes_dont_email_post_to_subs() {
...
+		// update a published post which should set the do not send flag
+		wp_update_post( array(
+			'ID'           => $new_post_id,
+			'post_content' => 'updated',
+			) );
+		
+		$this->sender->do_sync();
+
+		// revert back to what it used to be.
+		Jetpack_Options::update_option( 'active_modules', $active );
```
Comment (gravityrail) [inline tests/php/sync/test_class.jetpack-sync-posts.php]: Why is this line necessary?

File: tests/php/sync/test_class.jetpack-sync-posts.php [hunk 4]
```diff
@@ -364,6 +364,48 @@ function test_sync_post_includes_dont_email_post_to_subs() {
...
+			list( $post_id, $post ) = $event->args;
+			
+			if ( $post_id !== $new_post_id ) {
+				// ignore events that do not have the have the same post_id.
+				continue;
+			}
+
+			if ( 'updated' === $post->post_content || 'publish' !== $post->post_status ) {
+				// Only the updated post should have the do not send flag.
+				$this->assertEquals( true, $post->dont_email_post_to_subs, 'Post status:'. $post->post_status .' Should not send email to subscripbers.' );
```
Comment (gravityrail) [inline tests/php/sync/test_class.jetpack-sync-posts.php]: subscripbers => subscribers

File: tests/php/sync/test_class.jetpack-sync-posts.php [hunk 5]
```diff
@@ -364,6 +364,48 @@ function test_sync_post_includes_dont_email_post_to_subs() {
...
+		// create a draft
+		$new_post_id = $this->factory->post->create(  array( 'post_status' => 'draft' ) );
+
+		// Publish and then immediately update the post
+		wp_publish_post( $new_post_id );
+		// update a published post which should set the do not send flag
+		wp_update_post( array(
+			'ID'           => $new_post_id,
+			'post_content' => 'updated',
+			) );
```
Comment (ebinnion) [inline tests/php/sync/test_class.jetpack-sync-posts.php]: There's extra indentation here.

**Comment by gravityrail** [comment]
As discussed, rather than this solution, let's:

1) move the code which renders the `dont_email_post_to_subs` flag (and possibly similar flags based on meta) into the `jetpack_sync_before_enqueue_wp_insert_post` filter rather than the `jetpack_sync_before_send_wp_insert_post` so that the logic isn't based on stale data.

2) send a `jetpack_send_post_to_subscribers` event (similar to publicize) so that at some point we can adapt WPCOM to use that instead of the flag on the post.
